leader-election-impl/src/main/java/com/palantir/leader/PaxosLeaderElectionService.java

```diff
@@ -350,7 +352,7 @@ private void throwIfInvalidSetup(PingableLeader cachedService,
 
     @Override
     public Set<PingableLeader> getPotentialLeaders() {
-        return potentialLeadersToHosts.keySet();
+        return Sets.union(otherPotentialLeadersToHosts.keySet(), ImmutableSet.of(this));
```

Comment:
JavaDoc:

> Results are undefined if set1 and set2 are sets based on different equivalence relations (as HashSet, TreeSet, and the Map.keySet() of an IdentityHashMap all are).

I _think_ this is okay, especially given the tests pass, but might be good to have a double check.

nit: Also from JavaDoc

> Note: The returned view performs better when set1 is the smaller of the two sets. If you have reason to believe one of your sets will generally be smaller than the other, pass it first.

The first set should almost always be bigger, written this way (though the perf gain is obviously minimal, this is more of a random observation I made)

Reply:
I've switched the order around. We could make this a bit stronger by checking the UUIDs of the pingables and only adding ourselves if we're not in that list, but by design we aren't ever in that list anyway - so perhaps an additional test or two will be enough.

Reply:
Also since `otherPotentialLeadersToHosts` is final, it's not _possible_ for `this` to be in that list.